src/lib/context.ts

```diff
@@ -157,6 +158,24 @@ export const SourceDetector = {
   },
 };
 
+export const VariableTypeGuards = {
+  isHyperlink: function (variableData: VariableData): variableData is VariableHyperlink {
+    return "__type" in variableData && variableData.__type === "hyperlink";
+  },
+  isList: function (variableData: VariableData): variableData is VariableList {
+    return Array.isArray(variableData) && variableData.every((item) => typeof item === "string");
+  },
+  isMap: function (variableData: VariableData): variableData is VariableMap {
+    return "__type" in variableData && variableData.__type === "map";
+  },
+  isNumber: function (variableData: VariableData): variableData is VariableNumber {
+    return "__type" in variableData && variableData.__type === "number";
+  },
+  isString: function (variableData: VariableData): variableData is VariableString {
+    return "__type" in variableData && variableData.__type === "string";
+  },
+};
+
```

Comment:
It's worth noting that in the event the user is using `flat` or `icu` formats, we don't have access to any of the variable metadata, and that's something that we need to somehow account for

Reply:
Good point, I've updated the `VariableRenderPropType` to make the variable metadata optional. When using the flat or icu formats that data would be null and previous to this change we'd just return whatever input is passed into the variable. See the updated tests for examples of what that would look like in code.